client/src/hooks/graphql/use-with-review-answer-state.tsx

```diff
@@ -109,7 +109,7 @@ export function useWithReviewAnswerState(
           name: subject.name,
           description: subject.description,
           answers: uncategorizedAnswers,
-          recordAll: (status) => recordAnswers(status, subject._id, "none"),
+          recordAll: (status) => recordAnswers(status, subject._id, undefined),
```

Comment:
whenever possible avoid allowing or using `undefined` explicitly.

In this case, you have a variable that is a string type. You should almost always make the type strictly `string` and not `string | undefined`. Use the empty string to represent `unset`. Javascript makes this easy for you because the empty string is falsy, e.g

```typescript
s = ""
if(!s) { // this is true
}

Reply:
9382971